restapi_test.go

```diff
@@ -235,3 +237,41 @@ func Test_unmarshal(t *testing.T) {
 		t.Errorf("Unexpected error type: %T", err)
 	}
 }
+
+func TestWithContext(t *testing.T) {
+	// Set up a test context.
+	type key struct{}
+	ctx := context.WithValue(context.Background(), key{}, "foo")
```

Comment:
I think it's going to be better to replace it with `value`, to be more consistent with the naming.

```suggestion
	ctx := context.WithValue(context.Background(), key{}, "value")
```